netif.pyx

```diff
@@ -2037,18 +2048,29 @@ cdef list_interfaces_internal(names, typemap, defs.ifaddrs* ifa):
                 sin6 = <defs.sockaddr_in6*>ifa.ifa_dstaddr
                 addr.dest_address = ipaddress.ip_address(sin6.sin6_addr.s6_addr[:16])
 
-            ifd = <defs.if_data*>ifa.ifa_data
-            if ifd.ifi_vhid != 0:
-                addr.vhid = ifd.ifi_vhid
+        ifd = <defs.if_data*>ifa.ifa_data
+        if ifd.ifi_vhid != 0:
+            addr.vhid = ifd.ifi_vhid
 
         if sa.sa_family == defs.AF_LINK:
+            link = True
             if ifa.ifa_addr != NULL:
                 sdl = <defs.sockaddr_dl*>ifa.ifa_addr
                 nic.type = InterfaceType(sdl.sdl_type)
                 addr.address = LinkAddress(
                     sdl.sdl_data[:sdl.sdl_nlen],
                     ':'.join(['{0:02x}'.format(x) for x in bytearray(sdl.sdl_data[sdl.sdl_nlen:sdl.sdl_nlen+sdl.sdl_alen])]))
 
```

Comment:
Hvae you thought that you could put all this block within a `if link:` ? and drop a lot of these `if link else None`?